source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -78,14 +77,16 @@ void Filter::onDestroy() {
   }
 }
 
-void Filter::onComplete(Filters::Common::ExtAuthz::CheckStatus status) {
+void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
   ASSERT(cluster_);
-
   state_ = State::Complete;
 
   using Filters::Common::ExtAuthz::CheckStatus;
 
-  switch (status) {
+  const bool update_status_code = response->status_code >= enumToInt(Http::Code::Continue) &&
```

Comment:
Not sure I follow the logic here, can you add some comments?

Reply:
may be save the specific code we do want to use and base rest of the logic off it? (in case u think that will help)

Reply:
Agreed! That's super confusing and, I definitely change it. The idea is that when the authz service defines a new status code for "reject/denied", this logic will checks whether the code is valid and not equal to 403. If that evaluates to true, set a custom Status Code header and ResponseStatInfo. I will simplify this.